src/runtime_src/core/include/xcl_api_macros.h

```diff
@@ -32,22 +32,22 @@ mtx.unlock();
     AQUIRE_MUTEX()
 
 #define SERIALIZE_AND_SEND_MSG(func_name)\
-     unsigned c_len = c_msg.ByteSize(); \
+     unsigned c_len = c_msg.ByteSizeLong(); \
```

Comment:
Now that the return type is now known, shouldn't the variable type also be updated.  

unsigned is unfortunately not a good 'type', shouldn't it be instead unsigned long.  For example:
```suggestion
     unsigned long c_len = c_msg.ByteSizeLong(); \
```

Reply:
The code is probably wrong if the message size does not fit in an `int` like it was anyway wrong before if the message size was not fitting in the deprecated `ByteSize()`.
I do not have the big picture of the code so I do not know if it can happen or not.
Probably the code should get rid of macros, use more `auto` and functions/lambda with type-safety guarantees. But this sounds outside the scope of this...

Reply:
I just put `auto` to implement your suggestion.